src/set_theory/zfc/ordinal.lean

```diff
@@ -71,4 +76,65 @@ theorem is_transitive_iff_subset_powerset : x.is_transitive ↔ x ⊆ powerset x
 
 alias is_transitive_iff_subset_powerset ↔ is_transitive.subset_powerset _
 
+/-! ### Ordinals as sets -/
+
+/-- A set `x` is a von Neumann ordinal when it's a transitive set `x`, such that `y ∈ z ∈ w ∈ x`
+implies `y ∈ w`.
+
+There are many equivalences to this definition, which we aim to state and prove. These include:
+
+- A transitive set of transitive sets.
+- A hereditarily transitive set.
+- A transitive set that's transitive under `∈` (`is_ordinal_iff_is_trans`).
+- A transitive set that's trichotomous under `∈`.
+- A transitive set that's (strictly) totally ordered under `∈`.
+- A transitive set that's well-ordered under `∈`.
+-/
+def is_ordinal (x : Set) : Prop := x.is_transitive ∧ ∀ y z w : Set, y ∈ z → z ∈ w → w ∈ x → y ∈ w
```

Comment:
```suggestion
def is_ordinal (x : Set) : Prop := x.is_transitive ∧ ∀ w ∈ x, w.is_transitive
```
the definition here is just writing out what `is_transitive` means, which seems unnecessary.